# Eda on STEAM file

### importing important libraries

In [273]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [294]:
df = pd.read_csv(r"C:\Users\admin\Desktop\project\steam_project\src\steam_games_1000_final.csv")
df

title   price original_price discount  \
0                            Counter-Strike 2    Free           Free     Free   
1                     Umamusume: Pretty Derby    Free           Free     Free   
2                         PUBG: BATTLEGROUNDS    Free           Free     Free   
3                               Marvel Rivals    Free           Free     Free   
4                                        PEAK   359.0          359.0      0%%   
..                                        ...     ...            ...      ...   
995  Five Nights at Freddy's: Security Breach   899.0          899.0      0%%   
996                        Across the Obelisk   999.0          999.0      0%%   
997                   The Sims™ 4 Get To Work  2199.0         2199.0      0%%   
998           Two Point Museum: Fantasy Finds   665.0          665.0      0%%   
999                      MAVRIX by Matt Jones  1600.0         1600.0      0%%   

     release_date            platforms  \
0    21 Aug, 2012       Windows, Linux   
1    24 Jun, 2025              Windows   
2    21 Dec, 2017              Windows   
3     5 Dec, 2024              Windows   
4    16 Jun, 2025              Windows   
..            ...                  ...   
995  16 Dec, 2021              Windows   
996   8 Apr, 2021  Windows, Mac, Linux   
997  30 Mar, 2015              Windows   
998  17 Jul, 2025  Windows, Mac, Linux   
999  24 Jul, 2025              Windows   

                                         review_rating  review_count  
0    77% of the 78,097 user reviews in the last 30 ...     777809730  
1    93% of the 8,995 user reviews in the last 30 d...      93899530  
2    69% of the 15,651 user reviews in the last 30 ...     691565130  
3    56% of the 11,211 user reviews in the last 30 ...     561121130  
4    93% of the 41,839 user reviews in the last 30 ...     934183930  
..                                                 ...           ...  
995  84% of the 348 user reviews in the last 30 day...       8434830  
996  75% of the 37 user reviews in the last 30 days...        753730  
997  46% of the 260 user reviews for this game are ...         46260  
998  88% of the 36 user reviews in the last 30 days...        883630  
999  77% of the 390 user reviews for this game are ...         77390  

[1000 rows x 8 columns]

In [295]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 8 columns):
 #   Column          Non-Null Count  Dtype 
---  ------          --------------  ----- 
 0   title           1000 non-null   object
 1   price           1000 non-null   object
 2   original_price  1000 non-null   object
 3   discount        1000 non-null   object
 4   release_date    1000 non-null   object
 5   platforms       1000 non-null   object
 6   review_rating   1000 non-null   object
 7   review_count    1000 non-null   int64 
dtypes: int64(1), object(7)
memory usage: 62.6+ KB


## observation
- There are 7 object columns (title,price,origina_price,discount,release_date,platforms,review_rating)
- And 1 int column (review_count)

In [296]:
df.shape

(1000, 8)

## observation
- there are 1000 rows and 8 columns

In [297]:
df.columns

Index(['title', 'price', 'original_price', 'discount', 'release_date',
       'platforms', 'review_rating', 'review_count'],
      dtype='object')

# observation  : problem statment
1. Their are many games which are unknown and there specification is also null so now i will drop all the unknown rows.
2. In discount column there is % sign so we will remove it.
3. The release date column is object so we will change it to date and time
4. Price and original price column has free values so thats why that column is object so we will replce "free" with "0"
5. Discount column has "free" values so we will replace it with "100"
6. we will remove review column

In [298]:
df.duplicated().sum()

153

In [299]:
# Convert release_date to datetime
df['release_date'] = pd.to_datetime(df['release_date'], format='%d %b, %Y', errors='coerce')

# Check result
print(df['release_date'].head())
print(df['release_date'].dtype)


0   2012-08-21
1   2025-06-24
2   2017-12-21
3   2024-12-05
4   2025-06-16
Name: release_date, dtype: datetime64[ns]
datetime64[ns]


## observation 
- There are 153 duplicates so now i will drop them.

In [300]:
df = df.drop_duplicates().reset_index(drop = True)

In [301]:
df

title   price original_price discount  \
0                            Counter-Strike 2    Free           Free     Free   
1                     Umamusume: Pretty Derby    Free           Free     Free   
2                         PUBG: BATTLEGROUNDS    Free           Free     Free   
3                               Marvel Rivals    Free           Free     Free   
4                                        PEAK   359.0          359.0      0%%   
..                                        ...     ...            ...      ...   
842  Five Nights at Freddy's: Security Breach   899.0          899.0      0%%   
843                        Across the Obelisk   999.0          999.0      0%%   
844                   The Sims™ 4 Get To Work  2199.0         2199.0      0%%   
845           Two Point Museum: Fantasy Finds   665.0          665.0      0%%   
846                      MAVRIX by Matt Jones  1600.0         1600.0      0%%   

    release_date            platforms  \
0     2012-08-21       Windows, Linux   
1     2025-06-24              Windows   
2     2017-12-21              Windows   
3     2024-12-05              Windows   
4     2025-06-16              Windows   
..           ...                  ...   
842   2021-12-16              Windows   
843   2021-04-08  Windows, Mac, Linux   
844   2015-03-30              Windows   
845   2025-07-17  Windows, Mac, Linux   
846   2025-07-24              Windows   

                                         review_rating  review_count  
0    77% of the 78,097 user reviews in the last 30 ...     777809730  
1    93% of the 8,995 user reviews in the last 30 d...      93899530  
2    69% of the 15,651 user reviews in the last 30 ...     691565130  
3    56% of the 11,211 user reviews in the last 30 ...     561121130  
4    93% of the 41,839 user reviews in the last 30 ...     934183930  
..                                                 ...           ...  
842  84% of the 348 user reviews in the last 30 day...       8434830  
843  75% of the 37 user reviews in the last 30 days...        753730  
844  46% of the 260 user reviews for this game are ...         46260  
845  88% of the 36 user reviews in the last 30 days...        883630  
846  77% of the 390 user reviews for this game are ...         77390  

[847 rows x 8 columns]

In [302]:
df.duplicated().sum()

0

# observation
- Now we have droped duplicates.
- And with this now we have solved our 1 problem statment unknown rows are also droped now.

#### Now we are solving our next problem statment which is of 2 %% sign.

In [303]:
df['discount'] = df['discount'].str.replace(r'%%', '%', regex=True)

In [304]:
df

title   price original_price discount  \
0                            Counter-Strike 2    Free           Free     Free   
1                     Umamusume: Pretty Derby    Free           Free     Free   
2                         PUBG: BATTLEGROUNDS    Free           Free     Free   
3                               Marvel Rivals    Free           Free     Free   
4                                        PEAK   359.0          359.0       0%   
..                                        ...     ...            ...      ...   
842  Five Nights at Freddy's: Security Breach   899.0          899.0       0%   
843                        Across the Obelisk   999.0          999.0       0%   
844                   The Sims™ 4 Get To Work  2199.0         2199.0       0%   
845           Two Point Museum: Fantasy Finds   665.0          665.0       0%   
846                      MAVRIX by Matt Jones  1600.0         1600.0       0%   

    release_date            platforms  \
0     2012-08-21       Windows, Linux   
1     2025-06-24              Windows   
2     2017-12-21              Windows   
3     2024-12-05              Windows   
4     2025-06-16              Windows   
..           ...                  ...   
842   2021-12-16              Windows   
843   2021-04-08  Windows, Mac, Linux   
844   2015-03-30              Windows   
845   2025-07-17  Windows, Mac, Linux   
846   2025-07-24              Windows   

                                         review_rating  review_count  
0    77% of the 78,097 user reviews in the last 30 ...     777809730  
1    93% of the 8,995 user reviews in the last 30 d...      93899530  
2    69% of the 15,651 user reviews in the last 30 ...     691565130  
3    56% of the 11,211 user reviews in the last 30 ...     561121130  
4    93% of the 41,839 user reviews in the last 30 ...     934183930  
..                                                 ...           ...  
842  84% of the 348 user reviews in the last 30 day...       8434830  
843  75% of the 37 user reviews in the last 30 days...        753730  
844  46% of the 260 user reviews for this game are ...         46260  
845  88% of the 36 user reviews in the last 30 days...        883630  
846  77% of the 390 user reviews for this game are ...         77390  

[847 rows x 8 columns]

# observation 
- Our 2 problem is also solved i have removed 1 % sign from the data set

### now we are converting free values into 0 from price and original price column

In [305]:

# Replace "Free" with "0" in price and original_price columns
df["price"] = df["price"].astype(str).str.replace("Free", "0", case=False)
df["original_price"] = df["original_price"].astype(str).str.replace("Free", "0", case=False)

# Convert them to numeric (for calculations)
df["price"] = pd.to_numeric(df["price"], errors="coerce")
df["original_price"] = pd.to_numeric(df["original_price"], errors="coerce")


In [306]:
df

title   price  original_price  \
0                            Counter-Strike 2     0.0             0.0   
1                     Umamusume: Pretty Derby     0.0             0.0   
2                         PUBG: BATTLEGROUNDS     0.0             0.0   
3                               Marvel Rivals     0.0             0.0   
4                                        PEAK   359.0           359.0   
..                                        ...     ...             ...   
842  Five Nights at Freddy's: Security Breach   899.0           899.0   
843                        Across the Obelisk   999.0           999.0   
844                   The Sims™ 4 Get To Work  2199.0          2199.0   
845           Two Point Museum: Fantasy Finds   665.0           665.0   
846                      MAVRIX by Matt Jones  1600.0          1600.0   

    discount release_date            platforms  \
0       Free   2012-08-21       Windows, Linux   
1       Free   2025-06-24              Windows   
2       Free   2017-12-21              Windows   
3       Free   2024-12-05              Windows   
4         0%   2025-06-16              Windows   
..       ...          ...                  ...   
842       0%   2021-12-16              Windows   
843       0%   2021-04-08  Windows, Mac, Linux   
844       0%   2015-03-30              Windows   
845       0%   2025-07-17  Windows, Mac, Linux   
846       0%   2025-07-24              Windows   

                                         review_rating  review_count  
0    77% of the 78,097 user reviews in the last 30 ...     777809730  
1    93% of the 8,995 user reviews in the last 30 d...      93899530  
2    69% of the 15,651 user reviews in the last 30 ...     691565130  
3    56% of the 11,211 user reviews in the last 30 ...     561121130  
4    93% of the 41,839 user reviews in the last 30 ...     934183930  
..                                                 ...           ...  
842  84% of the 348 user reviews in the last 30 day...       8434830  
843  75% of the 37 user reviews in the last 30 days...        753730  
844  46% of the 260 user reviews for this game are ...         46260  
845  88% of the 36 user reviews in the last 30 days...        883630  
846  77% of the 390 user reviews for this game are ...         77390  

[847 rows x 8 columns]

# Observation
- our values are changed now

In [307]:
df.to_csv("cleaned_steam_games1000.csv", index=False, encoding="utf-8-sig")
print("Data saved as steam_games.csv")

Data saved as steam_games.csv


### Now we are replacing "free" values into 100% from discount column

In [308]:
# Replace "Free" with "100%" in discount column
df["discount"] = df["discount"].astype(str).str.replace("Free", "100%", case=False)
df

title   price  original_price  \
0                            Counter-Strike 2     0.0             0.0   
1                     Umamusume: Pretty Derby     0.0             0.0   
2                         PUBG: BATTLEGROUNDS     0.0             0.0   
3                               Marvel Rivals     0.0             0.0   
4                                        PEAK   359.0           359.0   
..                                        ...     ...             ...   
842  Five Nights at Freddy's: Security Breach   899.0           899.0   
843                        Across the Obelisk   999.0           999.0   
844                   The Sims™ 4 Get To Work  2199.0          2199.0   
845           Two Point Museum: Fantasy Finds   665.0           665.0   
846                      MAVRIX by Matt Jones  1600.0          1600.0   

    discount release_date            platforms  \
0       100%   2012-08-21       Windows, Linux   
1       100%   2025-06-24              Windows   
2       100%   2017-12-21              Windows   
3       100%   2024-12-05              Windows   
4         0%   2025-06-16              Windows   
..       ...          ...                  ...   
842       0%   2021-12-16              Windows   
843       0%   2021-04-08  Windows, Mac, Linux   
844       0%   2015-03-30              Windows   
845       0%   2025-07-17  Windows, Mac, Linux   
846       0%   2025-07-24              Windows   

                                         review_rating  review_count  
0    77% of the 78,097 user reviews in the last 30 ...     777809730  
1    93% of the 8,995 user reviews in the last 30 d...      93899530  
2    69% of the 15,651 user reviews in the last 30 ...     691565130  
3    56% of the 11,211 user reviews in the last 30 ...     561121130  
4    93% of the 41,839 user reviews in the last 30 ...     934183930  
..                                                 ...           ...  
842  84% of the 348 user reviews in the last 30 day...       8434830  
843  75% of the 37 user reviews in the last 30 days...        753730  
844  46% of the 260 user reviews for this game are ...         46260  
845  88% of the 36 user reviews in the last 30 days...        883630  
846  77% of the 390 user reviews for this game are ...         77390  

[847 rows x 8 columns]

# Observation
- our values are replaced now

### Now we are only keeping %values from review_rating column

In [309]:
df["review_rating"] = df["review_rating"].astype(str).str.extract(r"(\d+%)")
df

title   price  original_price  \
0                            Counter-Strike 2     0.0             0.0   
1                     Umamusume: Pretty Derby     0.0             0.0   
2                         PUBG: BATTLEGROUNDS     0.0             0.0   
3                               Marvel Rivals     0.0             0.0   
4                                        PEAK   359.0           359.0   
..                                        ...     ...             ...   
842  Five Nights at Freddy's: Security Breach   899.0           899.0   
843                        Across the Obelisk   999.0           999.0   
844                   The Sims™ 4 Get To Work  2199.0          2199.0   
845           Two Point Museum: Fantasy Finds   665.0           665.0   
846                      MAVRIX by Matt Jones  1600.0          1600.0   

    discount release_date            platforms review_rating  review_count  
0       100%   2012-08-21       Windows, Linux           77%     777809730  
1       100%   2025-06-24              Windows           93%      93899530  
2       100%   2017-12-21              Windows           69%     691565130  
3       100%   2024-12-05              Windows           56%     561121130  
4         0%   2025-06-16              Windows           93%     934183930  
..       ...          ...                  ...           ...           ...  
842       0%   2021-12-16              Windows           84%       8434830  
843       0%   2021-04-08  Windows, Mac, Linux           75%        753730  
844       0%   2015-03-30              Windows           46%         46260  
845       0%   2025-07-17  Windows, Mac, Linux           88%        883630  
846       0%   2025-07-24              Windows           77%         77390  

[847 rows x 8 columns]

# Observation 
- we have removed everything else rather than %

In [310]:
df.head()

title  price  original_price discount release_date  \
0         Counter-Strike 2    0.0             0.0     100%   2012-08-21   
1  Umamusume: Pretty Derby    0.0             0.0     100%   2025-06-24   
2      PUBG: BATTLEGROUNDS    0.0             0.0     100%   2017-12-21   
3            Marvel Rivals    0.0             0.0     100%   2024-12-05   
4                     PEAK  359.0           359.0       0%   2025-06-16   

        platforms review_rating  review_count  
0  Windows, Linux           77%     777809730  
1         Windows           93%      93899530  
2         Windows           69%     691565130  
3         Windows           56%     561121130  
4         Windows           93%     934183930

### we will remove review rating column

In [317]:
df = df.drop(columns=["review_rating"], errors="ignore")

# Observation 
- The column is now remove

In [318]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 847 entries, 0 to 846
Data columns (total 7 columns):
 #   Column          Non-Null Count  Dtype         
---  ------          --------------  -----         
 0   title           847 non-null    object        
 1   price           847 non-null    int64         
 2   original_price  847 non-null    int64         
 3   discount        847 non-null    int64         
 4   release_date    838 non-null    datetime64[ns]
 5   platforms       847 non-null    object        
 6   review_count    847 non-null    int64         
dtypes: datetime64[ns](1), int64(4), object(2)
memory usage: 46.4+ KB


# Observation
- we are done with our cleaning process

In [320]:
df.to_csv("steam_final2.csv",index = False)
